### PR TITLE
Feature: Adding VRP support for FDB table using HUAWEI-L2MAM-MIB

### PR DESCRIPTION
**Pull request by PipoCanaja**
Hello

Here is the code to load the FDB table using the Huawei MIB. Because the switch does not answer to Q-BRIDGE at all, and BRIDGE MIB does not provide the vlan information.

The test data (if necessary) will come after the other PR (POE polling) is merged, to have  a test file including all.

Bye

DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [X] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

### CODE REVIEW

File: includes/discovery/fdb-table/vrp.inc.php [hunk 1]
```diff
@@ -0,0 +1,47 @@
+<?php
+/**
+ * vrp.inc.php
+ *
+ * Discover FDB data with HUAWEI-L2MAM-MIB
```
Comment (laf) [inline includes/discovery/fdb-table/vrp.inc.php]: Could you update your copyright to use this template please:

https://docs.librenms.org/#Developing/Licensing/#copyright

File: includes/discovery/fdb-table/vrp.inc.php [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+                continue;
+            }
+            $vlan_id = isset($vlans_dict[$vlan]) ? $vlans_dict[$vlan] : 0;
+            $insert[$vlan_id][$mac_address]['port_id'] = $port_id;
+            d_echo("vlan $vlan mac $mac_address port ($ifIndex) $port_id\n");
+        }
+    }
+}
+
+echo PHP_EOL;
```
Comment (laf) [inline includes/discovery/fdb-table/vrp.inc.php]: Any reason for this? Probably better to add it to `echo 'HUAWEI-L2MAM-MIB:';` instead.

File: includes/discovery/fdb-table/vrp.inc.php (hunk 1 above)
Comment (PipoCanaja) [reply includes/discovery/fdb-table/vrp.inc.php]: Done

File: includes/discovery/fdb-table/vrp.inc.php (hunk 2 above)
Comment (PipoCanaja) [reply includes/discovery/fdb-table/vrp.inc.php]: Correct done. Validation ongoing.

**Comment by murrant** [comment]
Your test data files seem to be incomplete.
When I was attempting to generate the test data, the fdb data was not mapping to a port.

**Comment by murrant** [comment]
We don't support tests for the fdb-table module yet, added in #8723

**Comment by PipoCanaja** [comment]
@murrant OK.

How should I proceed then ? I wait until #8723 is merged ?

PipoCanaja

**Comment by PipoCanaja** [comment]
Hi @murrant and @laf 

Just a check to understand what's going on: the basic pre-commit test works but the advanced one (with test data) does not. When trying it on my machine, I get errors that do not seem to be related directly to this patch.So, the changes that are going on with the tests are impacting this merge ? 
If yes, could you let me know if there is something else I can do ? 
Thanx
PipoCanaja

**Comment by murrant** [comment]
@PipoCanaja It is a problem with the sql query to capture the data.  I need to update it but haven't had any type yet.

**Pull request merged**